cmd/cloudshell_open/cloudrun.go

```diff
@@ -32,10 +34,44 @@ const (
 	defaultRunMemory = "512Mi"
 )
 
+type service struct {
+	Spec serviceSpec `json:"spec"`
+	Status serviceStatus `json:"status"`
+}
+
+type serviceSpec struct {
+	Template serviceTemplate `json:"template"`
+}
+
+type serviceTemplate struct {
+	Spec templateSpec `json:"spec"`
+}
+
+type templateSpec struct {
+	Containers []container `json:"containers"`
+}
+
+type container struct {
+	EnvVars []envVar `json:"env"`
+}
+
+type envVar struct {
+	Name string `json:"name"`
+	Value string `json:"value"`
+}
+
+type serviceStatus struct {
+	Address address `json:"address"`
+}
+
+type address struct {
+	Url string `json:"url"`
+}
+
```

Comment:
you don't need these, just do it like this (inline anonymous type)


```
var v struct {
    Spec struct{
      Template struct { 
      } `json:"template"` 
    } `json:"spec"`
}
```

in the method that does unmarshal

Reply:
actually ignore the "inline" part.

it seems like you're not just using Url field. (but also env)
in that case, you can still create 1 type called Service, and kinda inline everything else into that type def.